sql/plan/fetch.go

```diff
@@ -18,74 +18,76 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/dolthub/go-mysql-server/sql/expression"
-	"github.com/dolthub/go-mysql-server/sql/types"
-
 	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
 )
 
 // Fetch represents the FETCH statement, which handles value acquisition from cursors.
 type Fetch struct {
-	Name      string
-	Variables []string
-	InnerSet  *Set
-	Pref      *expression.ProcedureReference
-	Sch       sql.Schema
+	Name     string
+	InnerSet *Set
+	ToSet    []sql.Expression
+	Pref     *expression.ProcedureReference
+	Sch      sql.Schema
 }
 
 var _ sql.Node = (*Fetch)(nil)
 var _ sql.CollationCoercible = (*Fetch)(nil)
 var _ expression.ProcedureReferencable = (*Fetch)(nil)
 
 // NewFetch returns a new *Fetch node.
-func NewFetch(name string, variables []string) *Fetch {
-	exprs := make([]sql.Expression, len(variables))
-	for i := range variables {
-		exprs[i] = expression.NewSetField(
-			expression.NewUnresolvedColumn(variables[i]),
-			expression.NewGetField(i, types.Null, "", true),
-		)
-	}
+func NewFetch(name string, toSet []sql.Expression) *Fetch {
 	return &Fetch{
-		Name:      name,
-		Variables: variables,
-		InnerSet:  NewSet(exprs),
+		Name:  name,
+		ToSet: toSet,
 	}
 }
 
 // Resolved implements the interface sql.Node.
 func (f *Fetch) Resolved() bool {
-	return f.InnerSet.Resolved()
+	for _, e := range f.ToSet {
+		if !e.Resolved() {
+			return false
+		}
+	}
+	return true
 }
 
 // String implements the interface sql.Node.
 func (f *Fetch) String() string {
-	return fmt.Sprintf("FETCH %s INTO %s", f.Name, strings.Join(f.Variables, ", "))
+	vars := make([]string, len(f.ToSet))
+	for i, e := range f.ToSet {
+		vars[i] = sql.DebugString(e)
```

Comment:
Should be normal String, maybe you want to implement the DebugString method instead